### PR TITLE
[terra-profile-image] Added Profile image hoc example

### PR DESCRIPTION
**Pull request by udaymattam**
### Summary

Added Higher order component example for Profile-image.

### Deployment Link

[https://terra-core-deployed-pr-#.herokuapp.com/
](https://terra-core-profile-imag-fowfu3.herokuapp.com/components/cerner-terra-core-docs/profile-image/about)
### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/profile-image/example/ProfileImagePlaceholderFallback.jsx [hunk 1]
```diff
@@ -7,11 +7,11 @@ const cx = classNames.bind(styles);
 
 const ProfileImagePlaceholderFallback = () => (
   <div>
-    <ProfileImage alt="profile image for Martin Moon, image failed to load" src="invalid.jpg" width="75" height="75" aria-describedby="caption-example-2" />
+    <ProfileImage alt="" src="invalid.jpg" width="75" height="75" aria-describedby="caption-example-2" />
```
Comment (supreethmr) [inline packages/terra-core-docs/src/terra-dev-site/doc/profile-image/example/ProfileImagePlaceholderFallback.jsx:10]: Curious to know why alt prop values are removed !!

File: packages/terra-profile-image/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@
 
 ## 3.41.1 - (January 4, 2022)
 
+* Changed 
+  * When `src` is not provided Aria-hidded is true.
```
Comment (saket2403) [inline packages/terra-profile-image/CHANGELOG.md:8]: Please change this so consumer can understand the impact of the change and move this to unreleased section.

File: packages/terra-core-docs/src/terra-dev-site/doc/profile-image/example/ProfileImageHigherOrderComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      </ProfileImage>
+      <React.Fragment>
+        <DropdownButton
+          label="Dropdown"
+          variant="ghost"
+        >
+          <Item label="Your Profile" onSelect={() => setMessage('Your in profile')} />
+          <Item label="Your organization" onSelect={() => setMessage('Your in organization')} />
+          <Item label="Your enterprise" onSelect={() => setMessage('Your in  enterprise')} />
+          <Item label="Your projects" onSelect={() => setMessage('Your in projects')} />
```
Comment (saket2403) [inline packages/terra-core-docs/src/terra-dev-site/doc/profile-image/example/ProfileImageHigherOrderComponent.jsx:31]: Nit:
```suggestion
          <Item label="Your Profile" onSelect={() => setMessage('You are in profile')} />
          <Item label="Your organization" onSelect={() => setMessage('You are in organization')} />
          <Item label="Your enterprise" onSelect={() => setMessage('You are in  enterprise')} />
          <Item label="Your projects" onSelect={() => setMessage('You are in projects')} />
```

File: packages/terra-profile-image/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,9 @@
 
 ## 3.41.1 - (January 4, 2022)
 
+* Changed 
+  * Minor dependency version bump
```
Comment (saket2403) [inline packages/terra-profile-image/CHANGELOG.md:8]: Change this entry to something meaningful wrt the changes and move it to the unreleased section.

**Comment by saket2403** [comment]
Can you add a jest test with no `alt` prop provided to the Profile image?

File: packages/terra-profile-image/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed 
+  * Added `role = presenetation` for Decorative profile-image.
```
Comment (sdadn) [inline packages/terra-profile-image/CHANGELOG.md:6]:
```suggestion
  * Added `role = presentation` for Decorative profile-image.
```

File: packages/terra-profile-image/src/ProfileImage.jsx [hunk 6]
```diff
@@ -82,7 +82,7 @@ const ProfileImage = (props) => {
   const profileImagePlaceholder = (
     <span
       {...customProps}
-      role="img"
+      role={alt ? 'img' : 'presenetation'}
```
Comment (sdadn) [inline packages/terra-profile-image/src/ProfileImage.jsx:85]:
```suggestion
      role={alt ? 'img' : 'presentation'}
```

File: packages/terra-profile-image/tests/jest/__snapshots__/ProfileImage.test.jsx.snap [hunk 7]
```diff
@@ -129,3 +129,16 @@ exports[`should render an avatar image without Source prop 1`] = `
   }
 />
 `;
+
+exports[`should render an avatar image without alt prop 1`] = `
+<span
+  className="profile-image placeholder cover default"
+  role="presenetation"
```
Comment (sdadn) [inline packages/terra-profile-image/tests/jest/__snapshots__/ProfileImage.test.jsx.snap:136]:
```suggestion
  role="presentation"
```

**Review by saket2403 — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**